src/corona_hakab_model/medical_state.py

```diff
@@ -9,12 +9,32 @@ class MedicalState(State, ABC):
     test_willingness: float
     detectable: bool
 
-    def __init__(self, *args, **kwargs):
-        self.susceptible = kwargs.pop('susceptible')
-        self.contagiousness = kwargs.pop('contagiousness')
-        self.test_willingness = kwargs.pop('test_willingness')
-        self.detectable = kwargs.pop('detectable')
-        super().__init__(*args, **kwargs)
+    def __init__(self, **kwargs):
+        self.susceptible = kwargs.get('susceptible')
+        self.contagiousness = kwargs.get('contagiousness')
+        self.test_willingness = kwargs.get('test_willingness')
+        self.detectable = kwargs.get('detectable')
+        super().__init__(kwargs.get('name'))
+
+    @staticmethod
+    def build(**kwargs):
+        """
+        Builds the appropriate MedicalState state object from a Dictionary
+        @param kwargs: dictionary of MedicalState parameters and additionally 'mechanism' which defines how the state
+        should function
+        @return: an instance of a MedicalState child class
+        """
+        medical_state_type = kwargs.pop('mechanism')
+        new_kwargs = {k: kwargs.get(k, None) for k in ("name",
+                                                       "susceptible",
+                                                       "contagiousness",
+                                                       "test_willingness",
+                                                       "detectable")}
```

Comment:
so someone can enter a completelt invalid JSON and you'll just return NONE (or possibly something worse?)

Reply:
No since the StateMachineBuilder will never allow that... but sure, I can handle this here too